pkg/controllers/jobset_controller.go

```diff
@@ -577,7 +577,9 @@ func (r *JobSetReconciler) createJobs(ctx context.Context, js *jobset.JobSet, jo
 		if err := r.Create(ctx, job); err != nil {
 			lock.Lock()
 			defer lock.Unlock()
-			finalErrs = append(finalErrs, fmt.Errorf("job %q creation failed with error: %v", job.Name, err))
+			err = fmt.Errorf("job %q creation failed with error: %w", job.Name, err)
+			finalErrs = append(finalErrs, err)
+			r.Record.Event(js, corev1.EventTypeWarning, constants.JobCreationFailedReason, err.Error())
```

Comment:
emit the event outside the parallel loop pls, the errors are already aggregated.